data/block/block.go

```diff
@@ -581,3 +582,39 @@ func (h *Header) GetAdditionalData() headerVersionData.HeaderAdditionalData {
 	// no extra data for the initial version of shard block header
 	return nil
 }
+
+// CheckFieldsForNil checks a predefined set of fields for nil values
+func (h *Header) CheckFieldsForNil() error {
+	if h == nil {
+		return data.ErrNilPointerReceiver
+	}
+	if h.PrevHash == nil {
+		return fmt.Errorf("%w in Header.PrevHash", data.ErrNilValue)
+	}
+	if h.PrevRandSeed == nil {
+		return fmt.Errorf("%w in Header.PrevRandSeed", data.ErrNilValue)
+	}
+	if h.RandSeed == nil {
+		return fmt.Errorf("%w in Header.RandSeed", data.ErrNilValue)
+	}
+	if h.PubKeysBitmap == nil {
```

Comment:
This field is nil for the proposed headers!

Reply:
Ok, will fix